src/nbl/video/CVulkanPhysicalDevice.h

```diff
@@ -1,88 +1,208 @@
+        for (uint32_t i = 0u; i < count; ++i)
+        {
+            result[i] = ISurfaceVK::getPresentMode(vk_presentModes[i]);
+        }
+
+        return result;
+    }
+    
+    uint32_t getMinImageCountForSurface(const ISurface* surface) const override
+    {
+        // Todo(achal): This is problematic, if passed `surface` isn't a vulkan surface
+        const ISurfaceVK* vk_surface = static_cast<const ISurfaceVK*>(surface);
+
+        VkSurfaceCapabilitiesKHR surfaceCapabilities;
+        vkGetPhysicalDeviceSurfaceCapabilitiesKHR(m_vkphysdev, vk_surface->m_surface,
+            &surfaceCapabilities);
+
+        return surfaceCapabilities.minImageCount;
+    }
+            
 protected:
     core::smart_refctd_ptr<ILogicalDevice> createLogicalDevice_impl(const ILogicalDevice::SCreationParams& params) override
     {
-        VkDeviceCreateInfo ci;
-        ci.sType = VK_STRUCTURE_TYPE_DEVICE_CREATE_INFO;
-        // TODO:
-        //ci.enabledExtensionCount
-        //ci.enabledLayerCount
-        //ci.pEnabledFeatures
-        //ci.ppEnabledExtensionNames
-        //ci.ppEnabledLayerNames
-        ci.pNext = nullptr;
-        ci.queueCreateInfoCount = params.queueParamsCount;
-
-        core::vector<VkDeviceQueueCreateInfo> qcis(ci.queueCreateInfoCount);
-        for (uint32_t i = 0u; i < qcis.size(); ++i)
+        VkDeviceCreateInfo createInfo = { VK_STRUCTURE_TYPE_DEVICE_CREATE_INFO };
+        // Todo(achal): Need to get this from the user
+        createInfo.enabledLayerCount = 1u;
+        const char* validationLayerName[] = { "VK_LAYER_KHRONOS_validation" };
+        createInfo.ppEnabledLayerNames = validationLayerName;
+
+        createInfo.queueCreateInfoCount = params.queueParamsCount;
+                
+        core::vector<VkDeviceQueueCreateInfo> queueCreateInfos(createInfo.queueCreateInfoCount);
+        for (uint32_t i = 0u; i < queueCreateInfos.size(); ++i)
         {
             const auto& qparams = params.queueCreateInfos[i];
-            auto& qci = qcis[i];
-
+            auto& qci = queueCreateInfos[i];
+                    
             qci.sType = VK_STRUCTURE_TYPE_DEVICE_QUEUE_CREATE_INFO;
             qci.pNext = nullptr;
             qci.queueCount = qparams.count;
             qci.queueFamilyIndex = qparams.familyIndex;
             qci.flags = static_cast<VkDeviceQueueCreateFlags>(qparams.flags);
             qci.pQueuePriorities = qparams.priorities;
         }
-        ci.pQueueCreateInfos = qcis.data();
-
+        createInfo.pQueueCreateInfos = queueCreateInfos.data();
+                
+        // Todo(achal): Need to get this from the user based on if its a headless compute or some presentation worthy workload
+        const uint32_t deviceExtensionCount = 1u;
+        const char* deviceExtensions[] = { VK_KHR_SWAPCHAIN_EXTENSION_NAME };
+                
+        createInfo.enabledExtensionCount = deviceExtensionCount;
+        createInfo.ppEnabledExtensionNames = deviceExtensions;
```

Comment:
Nabla has some sort of a rudimentary extension (more like capability) enum, you can refactor/adapt that
https://github.com/Devsh-Graphics-Programming/Nabla/blob/criss_android/include/IVideoCapabilityReporter.h#L27

Maybe split `SFeatures` into immutable features (just reported by the driver) vs enabled features (features you have to explicitly enable/ask for)
https://github.com/Devsh-Graphics-Programming/Nabla/blob/criss_android/include/nbl/video/IPhysicalDevice.h#L54